lib/facter/self_service.rb

```diff
@@ -108,6 +108,33 @@
     { S0021: Facter.value(:memory)['system']['capacity'].to_f <= 90 }
   end
 
+  chunk(:S0022) do
+    # Is there a valid license present, which does not expire in 90 days
+    # Also takes into account if the license is perpetual
+    next unless PuppetSelfService.primary?
+    validity = true # true by default
+    if File.file?('/etc/puppetlabs/license.key')
+      license_type = File.readlines('/etc/puppetlabs/license.key').grep(%r{license_type:}).to_s
+      if license_type.include? 'Perpetual'
+        validity = true
+      else
+        require 'date'
+        start_date = Date.parse(File.readlines('/etc/puppetlabs/license.key').grep(%r{start:}).to_s)
+        end_date = Date.parse(File.readlines('/etc/puppetlabs/license.key').grep(%r{end:}).to_s)
+        today_date = Date.today
+        daysexp = (end_date - today_date).to_i
+        validity = if (today_date > start_date) && (today_date < end_date) && (daysexp > 90)
```

Comment:
Great work! Here's a possible suggestion to get this into one line since the conditional only has two possible outcomes:

```
validity = (today_date > start_date) && (today_date < end_date) && (daysexp > 90) ? true : false
```

Reply:
Nice! I'll update this in my next commit, thank you